events/service/src/main/java/org/projectnessie/events/service/EventFactory.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.events.service;
+
+import jakarta.annotation.Nullable;
+import java.security.Principal;
+import java.util.Objects;
+import org.projectnessie.events.api.Content;
+import org.projectnessie.events.api.ContentKey;
+import org.projectnessie.events.api.Event;
+import org.projectnessie.events.api.EventType;
+import org.projectnessie.events.api.ImmutableCommitEvent;
+import org.projectnessie.events.api.ImmutableCommitMeta;
+import org.projectnessie.events.api.ImmutableContentRemovedEvent;
+import org.projectnessie.events.api.ImmutableContentStoredEvent;
+import org.projectnessie.events.api.ImmutableMergeEvent;
+import org.projectnessie.events.api.ImmutableReferenceCreatedEvent;
+import org.projectnessie.events.api.ImmutableReferenceDeletedEvent;
+import org.projectnessie.events.api.ImmutableReferenceUpdatedEvent;
+import org.projectnessie.events.api.ImmutableTransplantEvent;
+import org.projectnessie.events.service.util.ReferenceMapping;
+import org.projectnessie.versioned.BranchName;
+import org.projectnessie.versioned.Commit;
+import org.projectnessie.versioned.Hash;
+import org.projectnessie.versioned.MergeResult;
+import org.projectnessie.versioned.ReferenceAssignedResult;
+import org.projectnessie.versioned.ReferenceCreatedResult;
+import org.projectnessie.versioned.ReferenceDeletedResult;
+import org.projectnessie.versioned.ResultType;
+
+/**
+ * Factory for creating {@link Event}s from various version store objects, with all the boilerplate
+ * code that it requires.
+ *
+ * <p>This class is meant to be used as a singleton, or in CDI Dependent pseudo-scope.
+ */
+public class EventFactory {
+
+  protected final EventConfig config;
+
+  public EventFactory(EventConfig config) {
+    this.config = config;
+  }
+
+  protected Event newCommitEvent(
+      Commit commit, BranchName targetBranch, String repositoryId, @Nullable Principal user) {
+    org.projectnessie.model.CommitMeta commitMeta = commit.getCommitMeta();
+    ImmutableCommitEvent.Builder builder =
+        ImmutableCommitEvent.builder()
+            .id(config.getIdGenerator().get())
```

Comment:
I tried that initially, and while it makes our lives slightly easier when writing tests, I don't think these builders will be useful for end users, since they won't need to create event instances manually. That's why I decided to not include the builders.

Reply:
Okay.